### DISCUSSION TITLE
Question about intended behaviour when `soft-wrap` and `rulers` are enabled

### DISCUSSION THREAD

**Discussion by mtoohey31** [helix-editor/helix#6901] [General]
I'm wondering if the current behaviour when both `soft-wrap` and `rulers` are enabled is intentional. Here is what Helix and Neovim look like with `soft-wrap` enabled and `rulers`/`colorcolumn` in columns 60 and 80:

Helix:
![2023-04-19T23:12:02,117555697-04:00](https://user-images.githubusercontent.com/36740602/233249148-abfd357d-b85b-44d4-b56e-be31d33d7313.png)

Neovim:
![2023-04-19T23:12:04,820282466-04:00](https://user-images.githubusercontent.com/36740602/233249169-7cb39d42-53d7-4477-a0eb-cb617cd4dcc1.png)

In Helix's case, the column continues uninterrupted through the wrapped line, which does look nicer, but I'd argue the broken line displayed by Neovim is more functional, since the coloured background at the start of `vitae` accurately highlights the 80th character on that line.

So as stated at the top, my question is whether this behaviour is intentional. If this wasn't already explicitly decided upon, should we consider matching Neovim's behaviour here? Or is there another option that I haven't thought of which might be better than either one or the other? I suppose making this configurable would be an alternative, though I'm not sure what the performance impact would be.

Thoughts?

**Comment by kirawi** [comment]
I remember this being brought up in #5420 but I don't think it was implemented to keep the scope of the PR small.
